src/oop/object_independence/medium/q4/MainTest.java

```diff
@@ -0,0 +1,225 @@
+package oop.object_independence.medium.q4;
+
+import global.ObjectTest;
+import global.variables.Clause;
+import global.variables.clauses.StringLiteral;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertArrayEquals;
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class MainTest {
+    // Java
+    private final String className = "PortableMusicPlayer";
+    private ObjectTest testClass;
+    private final String firstFieldName = "playlist";
+    private final String secondFieldName = "current";
+    private final String firstMethod = "play";
+    private final String secondMethod = "playNext";
+    private final String thirdMethod = "playPrev";
+
+    @BeforeEach
+    public void setUp() {
+        String bankString = "oop.object_independence.medium.q4." + className;
+        testClass = new ObjectTest(bankString);
+    }
+
+    @Test
+    public void portableMusicPlayerClassHasRequiredAttributes() {
+        assertTrue(testClass.hasField(firstFieldName),
+                "Your " + className + " class is missing the " + firstFieldName + " attribute, or it is spelled incorrectly.");
+        assertTrue(testClass.hasField(firstFieldName, String[].class),
+                "Your " + firstFieldName + " attribute does not have the correct type.");
+        assertTrue(testClass.hasModifier(firstFieldName, "private"),
+                "Your " + firstFieldName + " attribute does not have the correct visibility modifier.");
+        assertTrue(testClass.hasField(secondFieldName),
+                "Your " + className + " class is missing the " + secondFieldName + " attribute, or it is spelled incorrectly.");
+        assertTrue(testClass.hasField(secondFieldName, int.class),
+                "Your " + secondFieldName + " attribute does not have the correct type.");
+        assertTrue(testClass.hasModifier(secondFieldName, "private"),
+                "Your " + secondFieldName + " attribute does not have the correct visibility modifier.");
+    }
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of((Object) new String[]{"Pineapple Juice", "Coconut Milk"}),
+                Arguments.of((Object) new String[]{"Music1", "Music2"}),
```

Comment:
You should add inputs with a greater variety of array lengths. There isn't really a point in testing two arrays of the same length, unless there is something about the strings in the array.

Reply:
We should test an array with length 1, 2, 3, and then a higher length like 24 or something.